api/messages/publish.proto

```diff
@@ -54,23 +59,23 @@ message DataStream {
 }
 
 message PublishReply {
- // Results describing messages successfully published to the queue.
- //
- // The output order matches the input order e.g. `results[N]` maps to `events[N]`,
- // however, `results` may contain only the first K results for `events`, when
- // not all the events are accepted by the queue, so the list sizes may not match.
- repeated EventResult results = 1;
-}
+ // The uuid of the shipper process, generated on startup. Clients can use this
+ // to detect when the shipper restarts.
+ string uuid = 1;
 
-message EventResult {
- // Timestamp of the event that was published.
- google.protobuf.Timestamp timestamp = 1;
- // ID derived from the queue offset that uniquely identifies the event within the shipper
- // system. Used to track the progress of events through the queue system.
- string queue_id = 2;
- // Optional. Event ID provided when the event was published, if it exists.
- string event_id = 3;
- // Optional. List of errors encountered by processors, if any. Processor
- // errors do not indicate the message failed to send.
- repeated google.rpc.Status processor_errors = 4;
+ // The number of events accepted by the shipper, in the same order as the
+ // PublishRequest. If uuid in the reply differs from uuid in the request,
+ // then accepted_count is always zero.
+ int32 accepted_count = 2;
+
+ // The final internal index for the events that were accepted. Inputs that
+ // want to guarantee event persistence can do it with this field: when the
+ // persisted_index of a PublishReply or InfoReply is >= this value, the
+ // events from this publish request have been persisted and the input can
+ // safely advance.
+ int64 accepted_index = 3;
```

Comment:
Let's reference the README for this parameter as well, or perhaps just on the top level message. The same goes for the `Info` message.